executor/common_linux.h

```diff
@@ -1466,6 +1466,98 @@ static long syz_extract_tcp_res(volatile long a0, volatile long a1, volatile lon
 #include "common_usb_linux.h"
 #endif
 
+#if SYZ_EXECUTOR || __NR_syz_raw_bind_info
+#include <linux/raw.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/sysmacros.h>
+static long syz_raw_bind_info(int fd, int arg)
+{
+	// syz_raw_bind_info$char_raw(dev fd, raw_dev int32[1:2]) block_info_ptr
+	// raw_dev is used to allow for multiple bindings per process.
+	struct stat sb;
+	int result = fstat(fd, &sb);
+	if (result == -1) {
+		return -1;
+	}
+	struct raw_config_request* rq = (struct raw_config_request*)malloc(sizeof(struct raw_config_request));
+	if (rq == NULL) {
+		return -1;
+	}
+	// +1 for /dev/raw/rawN, as N must be greater than 0
+	rq->raw_minor = procid + 1;
+#if SYZ_MULTI_PROC
+	rq->raw_minor += /*PROCS*/ *arg;
+#endif
+	rq->block_major = major(sb.st_rdev);
+	rq->block_minor = minor(sb.st_rdev);
+	debug("syz_raw_bind_info rq{%d (%d:%d)} fd: %d\n", (int)rq->raw_minor, (int)rq->block_major, (int)rq->block_minor, (int)fd);
+	return (long)rq;
+}
+#endif
+
+#if SYZ_EXECUTOR || __NR_syz_prepare_loop_device
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+static long syz_prepare_loop_device(int loopNo)
+{
+	int file_id = procid;
+#if SYZ_MULTI_PROC
+	file_id += /*PROCS*/ *loopNo;
+#endif
+
+	char tmpFileName[64];
+	snprintf(tmpFileName, sizeof(tmpFileName), "/tmp/fileRAW%d", file_id);
+	char loopname[64];
+	snprintf(loopname, sizeof(loopname), "/dev/loop%d", file_id);
+
+	int fd = open(tmpFileName, O_RDWR | O_CREAT | O_EXCL);
+	if (fd > 0) {
+		close(fd);
+		int status;
+		int pid = fork();
+		if (pid != 0) {
+			if (pid < 0) {
+				return -1;
+			}
+			waitpid(pid, &status, 0);
+			if (!WIFEXITED(status) || WEXITSTATUS(status) != 0) {
+				return -1;
+			}
+			debug("syz_prepare_loop_device executed dd\n");
+		} else {
+			char str[64];
+			strcpy(str, "of=");
+			strcat(str, tmpFileName);
+			execl("/bin/dd", "/bin/dd", "if=/dev/random", str, "bs=1024", "count=64", NULL);
+			return -1;
+		}
+	}
+
+	//[re]set loop device.
+	int status;
+	int pid = fork();
+	if (pid != 0) {
+		if (pid < 0) {
+			return -1;
+		}
+		waitpid(pid, &status, 0);
+		if (!WIFEXITED(status) || WEXITSTATUS(status) != 0) {
+			return -1;
+		}
+		debug("syz_prepare_loop_device executed losetup\n");
+	} else {
+		execl("/sbin/losetup", "/sbin/losetup", loopname, tmpFileName, NULL);
```

Comment:
I suspect losetup also does one or two syscalls that we really need. Run with strace to see what it's doing.

But also: can we do this (or some approximation) directly in descriptions?